### DISCUSSION TITLE
plane stress J2 plasticity - affine deformation - material properties at Gauss Points

### DISCUSSION THREAD

**Discussion by gsoim** [idaholab/moose#22888] [Q&A General] [answered]
Hi all,

I am trying to solve a simple 2D problem for two different material properties: a) plane stress elasticity and b) plane stress J2 plasticity. I would like to ask if it is possible to:
1) Define different material properties at each Gauss point, using elem_type=QUAD4 (i.e. Young's modulus for the linear elastic model or yield stress for the elasto-plastic model) and
2) Impose affine deformation on the boundaries in the input file (*.i) that uses a matrix multiplied by the coords at each boundary, that I already have as an input (practically it's the DefGrad that I already have from an upper-scale model analysis using Abaqus).
3) I have seen in the examples/tutorials several J2 plasticity problems for 3D, but I guess there is no problem for the plane stress case, right??

Thank you very much in advance.

Best,
George

**Comment by hugary1995** [comment, marked as answer]
> 1. Define different material properties at each Gauss point, using elem_type=QUAD4 (i.e. Young's modulus for the linear elastic model or yield stress for the elasto-plastic model) and

Yes, all material properties are defined per qp by default.

> 2. Impose affine deformation on the boundaries in the input file (*.i) that uses a matrix multiplied by the coords at each boundary, that I already have as an input (practically it's the DefGrad that I already have from an upper-scale model analysis using Abaqus).

Yes, use the Global Strain system or the Homogenization system.

> 3. I have seen in the examples/tutorials several J2 plasticity problems for 3D, but I guess there is no problem for the plane stress case, right??

We typically impose plane stress conditions weakly. Check out the Weak Plane Stress system.

There is a search button at the top left corner of the homepage. If that doesn't lead you to where you want, check out this page: https://mooseframework.inl.gov/modules/tensor_mechanics/index.html
